lib/sources/apollo/index.ts

```diff
@@ -0,0 +1,104 @@
+import BankModuleSource from '../bank-module';
+import { Logger } from 'pino';
+import { Tendermint34Client } from '@cosmjs/tendermint-rpc';
+import { CbOnUserBalances } from '../../../types/sources/cbOnUserBalances';
+import { queryContractOnHeight } from '../../query';
+import AstroportSource from '../astroport';
+
+export default class ApolloSource extends BankModuleSource {
+  rpc: string;
+  logger: Logger<never>;
+  assets: Record<string, { denom: string; vault_contract: string }>;
+  sourceName: string;
+  client: Tendermint34Client | undefined;
+
+  constructor(rpc: string, logger: Logger<never>, params: any) {
+    super(rpc, logger, params);
+    this.logger = logger;
+
+    if (!params.source) {
+      throw new Error('No source name configured in params');
+    }
+    this.sourceName = params.source;
+
+    if (!params.assets) {
+      throw new Error('No assets configured in params');
+    }
+    this.assets = params.assets;
+
+    this.rpc = rpc;
+  }
+
+  getUsersBalances = async (
+    height: number,
+    multipliers: Record<string, number>,
+    cb: CbOnUserBalances,
+  ): Promise<void> => {
+    const client = await this.getClient();
+    const astroportSource = new AstroportSource(this.rpc, this.logger, {
```

Comment:
If Apollo is not directly related to Astroport I would reimplement `getLpExchangeRate` there instead of instantiating astroport source

Reply:
it is. Like, Apollo uses Astroport for their vaults on Neutron